hibernate-core/src/test/java/org/hibernate/orm/test/metamodel/mapping/collections/SortNaturalByDefaultTests.java

```diff
@@ -0,0 +1,166 @@
+package org.hibernate.orm.test.metamodel.mapping.collections;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Tests Hibernate specific feature that {@link org.hibernate.annotations.SortNatural @SortNatural} will take effect implicitly
+ * when no <i>sort</i> or <i>order</i> related annotations exist, including:
+ * <ul>
+ *     <li>{@link org.hibernate.annotations.SortNatural @SortNatural}</li>
+ *     <li>{@link org.hibernate.annotations.SortComparator @SortComparator}</li>
+ *     <li>{@link org.hibernate.annotations.Sort @Sort}</li>
+ *     <li>{@link org.hibernate.annotations.OrderBy @OrderBy(from hibernate)}</li>
+ *     <li>{@link javax.persistence.OrderBy @OrderBy(from JPA)}</li>
+ * </ul>
+ * 
+ * @author Nathan Xu
+ */
+@ServiceRegistry
+@DomainModel(
+	annotatedClasses = {
+		SortNaturalByDefaultTests.Person.class,
+		SortNaturalByDefaultTests.Phone.class
+	}
+)
+@SessionFactory
+public class SortNaturalByDefaultTests {
+	
+	@Test
+	void test(SessionFactoryScope scope) {
+		scope.inTransaction(
+			session -> {
+				final Person person = session.createQuery( "select p from Person p", Person.class ).getSingleResult();
+				final SortedSet<Phone> phones = person.getPhones();
+				assertThat( phones.size(), is( 3 ) );
```

Comment:
We should also assert that the phones are in fact sorted in proper order.

Reply:
Updated. Originally I had thought we might end up with testing SortedSet or SortedMap. Testing case enriched enormously.